src/Node.cpp

```diff
@@ -23,29 +23,79 @@ namespace l3xz
 
 Node::Node()
 : rclcpp::Node("ros2_cyphal_bridge")
+, _node_heap{}
+, _node_hdl{_node_heap.data(),
+            _node_heap.size(),
+            [this] () { return micros(); },
+            [this] (CanardFrame const & frame) { return _can_mgr->transmit(frame); },
+            ::Node::DEFAULT_NODE_ID,
+            CYPHAL_TX_QUEUE_SIZE,
+            CYPHAL_RX_QUEUE_SIZE,
+            ::Node::DEFAULT_MTU_SIZE}
+, _node_mtx{}
+, _cyphal_node_start{std::chrono::steady_clock::now()}
+, _prev_heartbeat_timepoint{std::chrono::steady_clock::now()}
+, _prev_io_loop_timepoint{std::chrono::steady_clock::now()}
 {
+  _cyphal_heartbeat_pub = _node_hdl.create_publisher<uavcan::node::Heartbeat_1_0>(1*1000*1000UL /* = 1 sec in usecs. */);
+
+  _cyphal_node_info = _node_hdl.create_node_info(
+    /* uavcan.node.Version.1.0 protocol_version */
+    1, 0,
+    /* uavcan.node.Version.1.0 hardware_version */
+    1, 0,
+    /* uavcan.node.Version.1.0 software_version */
+    0, 1,
+    /* saturated uint64 software_vcs_revision_id */
+#ifdef CYPHAL_NODE_INFO_GIT_VERSION
+    CYPHAL_NODE_INFO_GIT_VERSION,
+#else
+    0,
+#endif
+    /* saturated uint8[16] unique_id */
+    std::array<uint8_t, 16>{0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F},
+    /* saturated uint8[<=50] name */
+    "107-systems.ros2_cyphal_bridge"
+  );
+
+  init_cyphal_to_ros_angle_actual();
+  init_cyphal_to_ros_tibia_endpoint_switch();
+  init_cyphal_to_ros_estop();
+  init_cyphal_to_ros_radiation_tick_cnt();
+  init_ros_to_cyphal_light_mode();
+
   declare_parameter("can_iface", "can0");
+  declare_parameter("can_node_id", 100);
+
+  RCLCPP_INFO(get_logger(),
+              "configuring CAN bus:\n\tDevice: %s\n\tNode Id: %ld",
+              get_parameter("can_iface").as_string().c_str(),
+              get_parameter("can_node_id").as_int());
 
-  RCLCPP_INFO(get_logger(), "configuring CAN bus:\n\tDevice: %s", get_parameter("can_iface").as_string().c_str());
+  _node_hdl.setNodeId(get_parameter("can_node_id").as_int());
 
-  _can_mgr = std::unique_ptr<CanManager>
-    (new CanManager(get_logger(),
-                    get_parameter("can_iface").as_string(),
-                    [this](CanardFrame const & frame)
-                    {
-                      RCLCPP_INFO(get_logger(), "CAN frame received");
-                    }));
+  _can_mgr = std::unique_ptr<CanManager>(new CanManager(
```

Comment:
This is a minor nit but static analyzers usually scream bloody murder when they see `new`, and for a good reason. I think a decent citizen would use a `std::make_unique` here or something like that.

Reply:
Huh, I did not know `std::make_unique` was a thing. Fixed in https://github.com/107-systems/ros2_cyphal_bridge/pull/45/commits/b4da6a3de553b5aab90be644bacd332c927b022f . Thank you for the input :bow: .